code/modules/mod/modules/modules_security.dm

```diff
@@ -361,13 +361,81 @@
 /obj/item/mod/module/active_sonar
 	name = "MOD active sonar"
 	desc = "Ancient tech from the 20th century, this module uses sonic waves to detect living creatures within the user's radius. \
+		Its basic function slowly scans around the user for any bio-signatures, however it can be overclocked to scan everywhere at once.\
 		Its loud ping is much harder to hide in an indoor station than in the outdoor operations it was designed for."
 	icon_state = "active_sonar"
 	module_type = MODULE_USABLE
-	use_power_cost = DEFAULT_CHARGE_DRAIN * 4
+	idle_power_cost = DEFAULT_CHARGE_DRAIN * 0.5
+	use_power_cost = DEFAULT_CHARGE_DRAIN * 3
 	complexity = 2
 	incompatible_modules = list(/obj/item/mod/module/active_sonar)
 	cooldown_time = 15 SECONDS
+	/// Time between us displaying radial scans
+	var/scan_cooldown_time = 1 SECONDS
+	/// The current slice we're going to scan
+	var/scanned_slice = 1
+	/// How many slices we make 360
+	var/radar_slices = 8 // 45 degrees each
+
+	/// A list of all creatures in range sorted by angle.
+	var/list/sorted_creatures = list()
+	/// A keyed list of all creatures
+	var/list/keyed_creatures = list()
+
+	/// Time between us displaying radial scans
+	COOLDOWN_DECLARE(scan_cooldown)
+
+/obj/item/mod/module/active_sonar/Initialize(mapload)
+	. = ..()
+	for(var/i in 1 to radar_slices)
+		sorted_creatures += list(list())
+
+/// Detects all living creatures within world.view, and returns the amount.
+/obj/item/mod/module/active_sonar/proc/detect_living_creatures()
+	var/creatures_detected = 0
+	for(var/mob/living/creature in range(world.view, mod.wearer))
+		if(creature == mod.wearer || creature.stat == DEAD)
+			continue
+		sort_creature_angle(creature)
+		RegisterSignal(creature, COMSIG_MOVABLE_MOVED, PROC_REF(sort_creature_angle), override = TRUE)
+		creatures_detected++
+	return creatures_detected
+
+/// Swaps around where a creature is
+/obj/item/mod/module/active_sonar/proc/sort_creature_angle(mob/living/creature, atom/old_loc, movement_dir, forced)
+	if(keyed_creatures[creature])
+		var/oldgroup = keyed_creatures[creature]
+		sorted_creatures[oldgroup] -= keyed_creatures
+	var/newgroup = round(get_angle(mod.wearer, creature) / (360 / radar_slices)) + 1
+	keyed_creatures[creature] = newgroup
+	sorted_creatures[newgroup] += creature
+
+/// Prunes creatures that aren't in our range anymore
+/obj/item/mod/module/active_sonar/proc/prune_creatures()
+	for(var/list/sublist in sorted_creatures)
+		for(var/mob/living/creature in sublist)
+			if(!get_dist(get_turf(mod.wearer), get_turf(creature)) > world.view)
```

Comment:
You already register for them moving, you could determine if they're out of range when they move instead of polling all of them in this proc. You'll need to detect when the user moves to detect distances too then.